libbeat/common/kubernetes/metadata.go

```diff
@@ -150,7 +150,7 @@ func generateMapSubset(input map[string]string, keys []string) common.MapStr {
 	for _, key := range keys {
 		value, ok := input[key]
 		if ok {
-			safemapstr.Put(output, key, value)
+			output[key] = value
```

Comment:
What happens if we would put it still in the same format?

Reply:
Do you mean with `Put`? `Put` creates the intermediate maps, in this case we want to use directly the labels as keys, no matter if they have dots or not.

Reply:
Yes, they can be used too, but this can be problematic, for example for these labels:
```
docker.container.labels.foo=something
docker.container.labels.foo.response=42
```
Using `Put` there can be two different possible results depending on what label is "put" first, this happens with the `json` type or without it, because this is caused by how we build the event:
```
{
  "docker": {
    "container": {
      "labels": {
        "foo": "something",
      }
    }
  }
}
``` 
Or
```
{
  "docker": {
    "container": {
      "labels": {
        "foo": {
          "response": "42"
        }
      }
    }
  }
}
``` 
With `safemapstr` this was resolved storing this like:
```
{
  "docker": {
    "container": {
      "labels": {
        "foo": {
          "response": "42",
          "value": "something"
        }
      }
    }
  }
}
``` 

Assigning directly and using the new `json` type, the labels keep their format in the stored event:
```
{
  "docker": {
    "container": {
      "labels": {
        "foo.response": "42",
        "foo": "something"
      }
    }
  }
}
```